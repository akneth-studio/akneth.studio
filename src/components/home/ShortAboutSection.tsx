import CTAButton from "../CTAButton";

export default function ShortAboutSection() {
  return (
    <section className="about-preview container bg-light py-5 px-4">
      <h2>O mnie</h2>
      <p>
        Nazywam się Katarzyna Pawłowska-Malesa. Jestem osobą otwartą, elastyczną i nie boję się żadnych wyzwań - zarówno w życiu, jak i w pracy. Programuję, tworzę szablony i automatyzacje, a każdemu projektowi poświęcam maksimum uwagi. Wierzę, że indywidualne podejście i szczerość to podstawa dobrej współpracy.
      </p>
      <CTAButton
        type='button'
        text="Dowiedz się więcej"
        variant="outline-primary"
        to="/about"
      />
    </section>
  );
}
