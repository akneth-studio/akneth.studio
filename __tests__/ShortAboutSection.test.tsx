import React from 'react';
import { render, screen } from '@testing-library/react';
import ShortAboutSection from '../src/components/home/ShortAboutSection';
import '@testing-library/jest-dom';

describe('ShortAboutSection component', () => {
  it('renders heading and short about text', () => {
    render(<ShortAboutSection />);
    // The heading text is "O mnie"
    expect(screen.getByRole('heading', { name: /o mnie/i })).toBeInTheDocument();
    // Add more specific checks if the component renders specific text or elements
  });

  it('matches snapshot', () => {
    const { container } = render(<ShortAboutSection />);
    expect(container).toMatchSnapshot();
  });
});
