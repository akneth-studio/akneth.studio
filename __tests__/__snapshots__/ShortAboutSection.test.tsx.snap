// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ShortAboutSection component matches snapshot 1`] = `
<div>
  <section
    class="about-preview container bg-light py-5 px-4"
  >
    <h2>
      O mnie
    </h2>
    <p>
      Nazywam się Katarzyna Pawłowska-Malesa. Jestem osobą otwartą, elastyczną i nie boję się żadnych wyzwań - zarówno w życiu, jak i w pracy. Programuję, tworzę szablony i automatyzacje, a każdemu projektowi poświęcam maksimum uwagi. Wierzę, że indywidualne podejście i szczerość to podstawa dobrej współpracy.
    </p>
    <button
      aria-label="Dowiedz się więcej"
      class="btn btn-outline-primary  btn btn-primary"
      data-testid="cta-button"
      role="button"
      tabindex="0"
      type="button"
    >
      Dowiedz się więcej
    </button>
  </section>
</div>
`;
